core/src/main/scala/pickling/Macros.scala

```diff
@@ -225,11 +229,16 @@ trait PicklerMacros extends Macro with PickleMacros with FastTypeTagMacros {
           else reflectively("picklee", fir)(fm => putField(q"$fm.get.asInstanceOf[${fir.tpe}]"))
         } else if (fir.javaSetter.isDefined) {
           List(putField(getField(fir)))
-        } else {
+        } else if (fir.isParam) {
           reflectivelyWithoutGetter("picklee", fir)(fvalue =>
             tryPutField(q"$fvalue.asInstanceOf[scala.util.Try[${fir.tpe}]]"))
+        } else {
+          Nil
```

Comment:
what does this case mean exactly, that we found a field but don't know how to pickle it? If so should the error be here (if we're losing _any_ field it's an error) rather than below (if we lose _all_ the fields)?